quickevent/app/plugins/Competitors/src/competitorswidget.h

```diff
@@ -40,18 +40,16 @@ class CompetitorsWidget : public QFrame
 	void editCompetitors(int mode);
 
 	void editCompetitor_helper(const QVariant &id, int mode, int siid);
-
-	void onDbEventNotify(const QString &domain, int connection_id, const QVariant &data);
-	void editCompetitorOnPunch(int siid);
+	Q_SLOT void editCompetitorOnPunch(int siid);
 
 	void onCustomContextMenuRequest(const QPoint &pos);
 
 	void report_competitorsStatistics();
 private:
+	bool editCompetitorLock = false;
```

Comment:
`m_editCompetitorLock`, but I do not like locks as member fields. They are left unlocked in case of exception and app stop to work. Try to make lock-free implementation or use some kind of scope guard to avoid stay-locked-on-exception

Reply:
ps: previous implementation had exactly this functionality, but I will look into it.

Reply:
but checkbox can be explicitly unlocked by user because its state is visible and controllable in UI

Reply:
truth, anyway solved using RAII mutex